app/controllers/hyrax/handles_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Hyrax
+  class HandlesController < ApplicationController
+    def handle_show
+      if params[:handle_id] != "1957"
+        puts "Handle ID does not match with Scholars Archive Handle ID."
+      elsif params[:id].length != 5
+        puts "ID is not the right length for Scholars Archive Handle ID."
+      end
+
+      solr_doc = query_solr_for_work(construct_handle_url(params[:id]))
+      work = query_fedora_for_work(solr_doc.first[:id], solr_doc.first[:human_readable_type_tesim].first.constantize)
+      redirect_to "/concern/#{work.class.to_s.downcase}s/#{work.id}"
+    end
+
+    def handle_download
+    end
+
+    private
+      def construct_handle_url(handle_code)
+        "hdl.handle.net/1954/" + handle_code
```

Comment:
since you have reference to it, you should take in and use the `handle_id` param.. avoid hardcoding